syntax = "proto3";
option cc_enable_arenas = true;

// this aims to be somewhat compatible with ga4gh
// protobuf schemas -- experimental

message Position {
  // position within the contig
  int64 position = 1;
  // which contig in the reference (name and index)
  int32 ref_index = 2;
  string contig = 3;
}

message Alignment {
  uint32 flag = 1;
  int32 mapping_quality = 2;
  Position position = 3;
  Position next_position = 4;
  int64 template_length = 5;
  string cigar = 6;
}

